src/Spin Button/main.vala

```diff
@@ -0,0 +1,40 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+string tell_time (Gtk.SpinButton hours, Gtk.SpinButton minutes) {
+    return (@"The time selected is $(hours.text):$(minutes.text)");
+}
+
+public void main () {
+    var hours = (Gtk.SpinButton) workbench.builder.get_object ("hours");
+    var minutes = (Gtk.SpinButton) workbench.builder.get_object ("minutes");
+
+    hours.text = "00";
+    minutes.text = "00";
+
+    hours.value_changed.connect (() => {
+        message (@"$(tell_time (hours, minutes))");
+    });
+
+    minutes.value_changed.connect (() => {
+        message (@"$(tell_time (hours, minutes))");
+    });
+
+
+    hours.output.connect (() => {
+        double value = hours.adjustment.value;
```

Comment:
Cast the value to `int` in this variable instead of doing it in the formatting, applies to the other signal handler too.
```suggestion
        var value = (int) hours.adjustment.value;
```